File: templates/stacks/python/benchmark.py
"""
Forge benchmark fixture — Python stack.
Run via: pytest benchmark.py --benchmark-only

Fill in the `target_operation()` function with the code path you want to measure.
The PERF_SCORE in EVAL.sh will use the median runtime of this fixture.

Install: pip install pytest-benchmark
"""
import pytest


def target_operation():
    """
    Replace this with the operation you want to benchmark.
    Example: parse a file, run a calculation, call a local function.
    Keep it representative of a typical request/operation in your project.
    """
    # TODO: replace with your actual operation
    total = sum(range(10_000))
    return total


def test_benchmark_target(benchmark):
    """Benchmark fixture — PERF_SCORE is derived from this."""
    result = benchmark(target_operation)
    # Basic sanity check — adjust as needed
    assert result is not None

File: templates/stacks/node/benchmark.js
/**
 * Forge benchmark fixture — Node stack.
 * Run via: npx vitest bench (requires vitest >=0.34)
 *
 * Fill in targetOperation() with the code path you want to measure.
 * The PERF_SCORE in EVAL.sh will use the ops/sec from this fixture.
 *
 * Install: npm install --save-dev vitest
 */
import { bench, describe } from "vitest";

function targetOperation() {
  // TODO: replace with your actual operation.
  // Example: parse JSON, run a calculation, call a local function.
  let total = 0;
  for (let i = 0; i < 10_000; i++) total += i;
  return total;
}

describe("Forge PERF benchmark", () => {
  bench("target operation", () => {
    targetOperation();
  });
});
